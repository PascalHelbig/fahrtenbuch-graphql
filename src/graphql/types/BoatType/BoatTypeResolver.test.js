const { makeExecutableSchema } = require('graphql-tools');
const { graphql } = require('graphql');
const BoatType = require('./BoatType');
const BoatTypeResolver = require('./BoatTypeResolver');

const testPropterty = (propterty, returnValue) => {
  const SchemaDefinition = `
    type Query {
      boat: Boat
    }
        
    schema { query: Query }`;

  const mockBookshelf = {
    get: jest.fn().mockReturnValue(returnValue),
    related: jest.fn().mockReturnValue(returnValue),
  };
  const resolvers = {
    Query: {
      boat: () => mockBookshelf,
    },
    Boat: BoatTypeResolver,
  };
  const schema = makeExecutableSchema({ typeDefs: [SchemaDefinition, BoatType], resolvers });

  const query = `
    query { 
      boat {
        ${propterty}
     }
    }
  `;

  return graphql(schema, query).then((result) => {
    expect(result).toMatchSnapshot();
    return mockBookshelf;
  });
};

it('should get the id from model', () =>
  testPropterty('id', '1337').then(({ get }) =>
    expect(get).toHaveBeenCalledWith('id')
  )
);

it('should get the name from model', () =>
  testPropterty('name', 'the name').then(({ get }) =>
    expect(get).toHaveBeenCalledWith('name')
  )
);

it('should get the owner from model', () =>
  testPropterty('owner { id }', { id: 'the owner' }).then(({ related }) =>
    expect(related).toHaveBeenCalledWith('owner')
  )
);
