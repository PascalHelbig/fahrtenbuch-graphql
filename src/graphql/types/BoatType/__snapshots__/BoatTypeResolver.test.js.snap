exports[`test should get the id from model 1`] = `
Object {
  "data": Object {
    "boat": Object {
      "id": "1337",
    },
  },
}
`;

exports[`test should get the name from model 1`] = `
Object {
  "data": Object {
    "boat": Object {
      "name": "the name",
    },
  },
}
`;

exports[`test should get the owner from model 1`] = `
Object {
  "data": Object {
    "boat": Object {
      "owner": null,
    },
  },
  "errors": Array [
    [GraphQLError: Generated Schema cannot use Interface or Union types for execution.],
  ],
}
`;
